test/html_to_json/Html_to_jsonTest.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package html_to_json;
+
+import java.io.IOException;
+import static jdk.nashorn.internal.objects.Global.getJSON;
+import net.sf.json.JSONSerializer;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.jsoup.Connection;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import static org.junit.Assert.*;
+import org.skyscreamer.jsonassert.JSONAssert;
+
+/**
+ *
+ * @author FireHawk2300
+ */
+public class Html_to_jsonTest {
+    
+    public Html_to_jsonTest() throws IOException {
+        
+        String html = Jsoup.connect("https://github.com/egis/handbook/blob/master/Tech-Stack.md").get().html();
+        
+        Document doc = Jsoup.parse(html);
+        
+    }
+    // Method used to verify JSON format
+    // Returns boolean result
+    public static boolean isValidJSON(String test) {
+        try {
+            new JSONObject(test);
+        } catch (JSONException ex) {
+            // edited, to include @Arthur's comment
+            // e.g. in case JSONArray is valid as well...
+            try {
+                new JSONArray(test);
+            } catch (JSONException ex1) {
+                return false;
+            }
+        }
+        return true;
+    }
+    
+    @Test
+    public void checkConn() throws IOException {
+        //Tests connection to url
+        Connection conn = Jsoup.connect("https://github.com/egis/handbook/blob/master/Tech-Stack.md"); 
+        conn.get(); // would throw exception on error
+        assertTrue(true);
+    }
+    
+    
+    @Test
+    public void checkRedirects() {
+        //Tests for stable connection
+        boolean conn = true;
+        try {
+            Document doc = Jsoup.connect("https://github.com/egis/handbook/blob/master/Tech-Stack.md").get();
+        } catch (IOException e) {
+            assertTrue(e.getMessage().contains("Too many redirects"));
+            conn = false;
+        }
+        assertTrue(conn);
```

Comment:
If you're checking for redirects, why are asserting if the connection is made or not? This is not part of this test.